hazelcast-tpc-engine/src/main/java/com/hazelcast/internal/tpc/nio/NioAsyncSocket.java

```diff
@@ -137,7 +158,103 @@ public void setWriteThrough(boolean writeThrough) {
     }
 
     @Override
-    public void soLinger(int soLinger) {
+    public void setTcpKeepAliveTime(int keepAliveTime) {
+        checkPositive(keepAliveTime, "keepAliveTime");
+        if (TCP_KEEPIDLE == null) {
+            if (TCP_KEEPIDLE_PRINTED.compareAndSet(false, true)) {
+                logger.warning("Ignoring NioAsyncSocket.setTcpKeepAliveTime. " +
+                        "Please upgrade to Java 11+ or configure tcp_keepalive_time in the kernel. " +
+                        "For more info see https://tldp.org/HOWTO/html_single/TCP-Keepalive-HOWTO/. " +
+                        "If this isn't dealt with, idle connections could be closed prematurely.");
```

Comment:
Effectively the keep-alive functionality is useless against firewalls/routers closing connections due to inactivity because the interval of sending fake packets is way too high (7200 seconds).